src/c11threads.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _C11_THREADS_H_
+#define _C11_THREADS_H_
+
+
+/* If there's C11 <threads.h> support, use it; otherwise use tinycthread.h,
+ * which has a compatible API. */
+#ifdef THREADS_H_SUPPORT
```

Comment:
I think I'd rather see this require `THREADS_H_SUPPORT` to be defined one way or another, so we know that the build environment made a conscious choice rather than not knowing this issue exists. (For example, Makevars.win doesn't deal with this yet, but it would happily compile anyway...)